cloudinit/config/cc_ssh.py

```diff
@@ -278,7 +278,7 @@ def handle(
                     if gid != -1:
                         # perform same "sanitize permissions" as sshd-keygen
                         os.chown(keyfile, -1, gid)
-                        os.chmod(keyfile, 0o640)
+                        os.chmod(keyfile, 0o600)
```

Comment:
This was introduced in https://github.com/canonical/cloud-init/commit/ee296ced9c0a61b1484d850b807c601bcd670ec1, and the read group permissions are specific to the `ssh_keys` group in system with that group set (RHEL-like OSes).

I would assume that that fact is a requirement needed for `openscap` to function properly. Maybe @esposem can clarify that.

In any case, I think it is safe to keep this line as it was (`0o640`) as only users belonging to the `ssh_keys` group would be able to read private keys, which makes sense.

Reply:
The issue is that we are replacing sshd-keygen. So cloud-init should do the same exact thing that sshd-keygen does.
And when reading `/usr/libexec/openssh/sshd-keygen` this is what we find (also posted in my PR):

```
# create new keys
if ! $KEYGEN -q -t $KEYTYPE -f $KEY -C '' -N '' >&/dev/null; then
	exit 1
fi

# sanitize permissions
/usr/bin/chgrp ssh_keys $KEY
/usr/bin/chmod 640 $KEY
/usr/bin/chmod 644 $KEY.pub
if [[ -x /usr/sbin/restorecon ]]; then
	/usr/sbin/restorecon $KEY{,.pub}
fi
```
That's why I wrote `640` and not `600`.

Reply:
good to know.

that is also fine for me. Will change this back to 0640.

Should we then also change this to 640?

https://github.com/canonical/cloud-init/blob/e49852d44aba7e32e763244af2cb152b214b2306/cloudinit/config/cc_ssh.py#L189

Reply:
Thanks @Mazorius. I think line 189 does not interfere with the `ssh_keys` case, and it is right as-is.

Reply:
Agree

Reply:
changed back to 640